wordpress-mobile/WordPress-Android
# Comments snippet under post.

Part of #15660

There are a few files involved in this PR but nearly half of them are simple ViewHolders and layout related files, so hopefully it's good enough to review.

This PR does the following:
 - Introduces a COMMENTS_SNIPPET feature flag (that must be activated to test the feature)
 - Introduces the comments snippet below post content. This shows first comment if present and gives access to view all comments. Also `follow conversation` is possible from here.

A few special cases to consider:

| Post with comments and comments are closed: no Follow Conversation present | ![image](https://user-images.githubusercontent.com/47797566/145261074-c35fd5f9-48aa-4aa0-b1f0-28007645a480.png) |
|---|---|
| Post with comments and comments opened: Follow Conversation present | ![image](https://user-images.githubusercontent.com/47797566/145261226-e9d0d361-21cb-44ef-b686-2fadcfff24fa.png) |
| Post without comments and comments are closed: no Follow Conversation and closed comment label | ![image](https://user-images.githubusercontent.com/47797566/145261372-b0e1246e-1be5-427e-900b-3f499e028670.png) |
| Post without comments and comments opened: Follow Conversation and No comments yet label | ![image](https://user-images.githubusercontent.com/47797566/145261658-e3268485-ac27-4bb3-a226-c80c1b71a7a9.png) |


There are a few bits that can be unit tested but unit test is being added in a separate PR. Also tracking is being added in a separate PR.

## To test
- Check you do not have the comment snippet when the FF is off and it's there when the FF is on
- Check the special cases reported in the images above
- Check you can open the threaded comment view by tapping on the button (View all comments or Be the first to comment depending on the scenario)
- Check you can Follow/Unfollow by email or by push notification. Smoke test the feature (should behave the same as in the threaded comments)
- Try to rotate the screen and check no visual issues happen
- Smoke test the Post Detail screen (follow site, like post, save for later, whatever you can thing ok :) )

## Regression Notes
1. Potential unintended areas of impact
The feature is still behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
The feature is still behind a feature flag.

3. What automated tests I added (or what prevented me from doing so)
The feature is still behind a feature flag and a few more automated tests are being submitted with a dedicated PR.

PR submission checklist:
